Meadowlark-js/backends/meadowlark-mongodb-backend/src/repository/Delete.ts

```diff
@@ -70,8 +70,10 @@ export async function deleteDocumentById(
       // Perform the document delete
       Logger.debug(`mongodb.repository.Delete.deleteDocumentById: Deleting document id ${id}`, traceId);
 
-      const { deletedCount } = await mongoCollection.deleteOne({ _id: id }, { session });
-      deleteResult = deletedCount === 0 ? { response: 'DELETE_FAILURE_NOT_EXISTS' } : { response: 'DELETE_SUCCESS' };
+      const { acknowledged, deletedCount } = await mongoCollection.deleteOne({ _id: id }, { session });
+      if (acknowledged) {
```

Comment:
This was the only area I found to be lacking. If `acknowledged` is false, the rest of the write operation results come back as `undefined`, which is not well expressed in the TypeScript types.